test/metabase/task/send_pulses_test.clj

```diff
@@ -208,12 +209,13 @@
                                                                 :channel_type   :slack
                                                                 :details        {:channel "#random"}}
                                                                daily-at-6pm)))]
-              (testing "sanity check that we have the correct number triggers and no channel has been sent yet"
-                (is (= pc-count (->> pulse-ids (map pulse-channel-test/send-pulse-triggers) (apply set/union) count)))
-                (is (= #{} @sent-channel-ids)))
-              ;; job run every 2 seconds, so wait a bit so the job can run
-              (Thread/sleep 3000)
+              (testing "sanity check that we have the correct number of triggers and no channel has been sent yet"
+                (is (= pc-count (->> pulse-ids (map pulse-channel-test/send-pulse-triggers) (apply set/union) count))))
```

Comment:
The previous check is `(is (= #{} @sent-channel-ids))` is not needed; all we want to make sure here is that if there are more jobs than the number of available threads that are scheduled at the same time, then quartz won't drop any of it.